### PR TITLE
Updated TiledRasterLayer's Reproject Method

### PR DESCRIPTION
**Pull request by jbouffard**
This PR updates the `reproject` method for `TiledRasterLayer` by having it accept various values for the `layout` including `LocalLayout` and `GlobalLayout`.

### CODE REVIEW

File: geopyspark/geotrellis/layer.py [hunk 1]
```diff
@@ -892,54 +892,61 @@ def convert_data_type(self, new_type, no_data_value=None):
             return TiledRasterLayer(self.pysc, self.layer_type,
                                     self.srdd.convertDataType(CellType(new_type).value))
 
-    def reproject(self, target_crs, extent=None, layout=None, scheme=LayoutScheme.FLOAT, tile_size=256,
-                  resolution_threshold=0.1, resample_method=ResampleMethod.NEAREST_NEIGHBOR):
+    def reproject(self, target_crs, layout=LocalLayout(),
```
Comment (echeipesh) [inline geopyspark/geotrellis/layer.py]: Default here should be `None`, trying to match the layout there is.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala [hunk 2]
```diff
@@ -100,31 +100,25 @@ abstract class TiledRasterRDD[K: SpatialComponent: JsonFormat: ClassTag] extends
...
-    tileSize: Int,
-    resolutionThreshold: Double,
+    layoutType: LayoutType,
     crs: String,
     resampleMethod: ResampleMethod
   ): TiledRasterRDD[K] = {
-    val _crs = TileRDD.getCRS(crs).get
+    val convertedCRS = TileRDD.getCRS(crs).get
+    val layoutDefinition =
+      layoutType.layoutDefinition(convertedCRS, rdd.metadata.extent, rdd.metadata.cellSize)
```
Comment (jbouffard) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala]: Would doing `rdd.metadata.extent.reproject(convertedCRS)` work?

Comment (echeipesh) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala]: No, that won't do the trick by itself. The reprojection could bow the extent where just reproejcting the envelope won't actually cover all the reprojected cells. I think we may have densified the envelope to reproject it in the past, but I'll have to double check that.

Comment (jbouffard) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala]: I see. In the meantime then, I fix the python code.

**Comment by jbouffard** [comment]
Superseded by #390

**Pull request closed without merging**